python/cudf/cudf/core/index.py

```diff
@@ -2257,7 +2257,7 @@ def days(self):
         Number of days for each element.
         """
         return as_index(
-            arbitrary=self._values.days, name=self.name, dtype="int32"
+            arbitrary=self._values.days, name=self.name, dtype="int64"
```

Comment:
Why does only this one need changed to int64?

Reply:
We had to change just for `days` because it is not constrained like the rest of parameters in the range of values and there is an overflow happening:
```python
In [1]: import cudf

In [2]: i = cudf.Index([245345345545332], dtype='timedelta64[s]')

In [3]: i.days
Out[3]: Index([2839645203], dtype='int64')

In [4]: i
Out[4]: TimedeltaIndex(['2839645203 days 01:42:12'], dtype='timedelta64[s]')

In [5]: i.days
Out[5]: Index([-1455322093], dtype='int32')          # Overflow prior to this change.
```

Reply:
Aha, thanks! Maybe add a comment?

Reply:
Done 👍